### PR TITLE
#898: Bugfix for aws commandlet

### PR DESCRIPTION
**Pull request by CREITZ25**
Bugfix for aws commandlet

### CODE REVIEW

File: scripts/src/main/resources/scripts/command/aws [hunk 1]
```diff
@@ -16,8 +16,22 @@ source "$(dirname "${0}")"/../functions
 
 # Call aws with specified arguments.
 function doRun() {
-  doSetup silent
```
Comment (hohwille) [inline scripts/src/main/resources/scripts/command/aws:19]: why is this not wroking?
This is the common way to do it in all other commandlets.
Cant we make it work the same way here?

File: scripts/src/main/resources/scripts/command/aws [hunk 2]
```diff
@@ -29,43 +43,46 @@ function doPackageInstall()
...
+      fi
+
+      if [ "${1}" != "silent" ] && ! doIsQuiet
+      then
+        doEcho "logon again to test aws with \"aws --version\""
+      fi 
     fi
+  else
+    echo "aws is already installed"
+    doRunCommand "aws --version"
```
Comment (hohwille) [inline scripts/src/main/resources/scripts/command/aws:82]: `silent` means that if the tool is already is installed, then nothing should be logged and almost nothing should happen (that takes time or requires network access). So this `else` block should only happen if not silent.
The regular thing to do that in all other commandlets is that separate `if` that you removed here:
```
if [ "${1}" != "silent" ] && ! doIsQuiet
```
You can simply remove the message "aws is already installed".

File: scripts/src/main/resources/scripts/command/aws [hunk 3]
```diff
@@ -29,43 +43,46 @@ function doPackageInstall()
...
+    powershell.exe -Command "Start-Process msiexec.exe -verb runas -Wait -ArgumentList '/I ${windows_path_to_package}\aws-${AWS_VERSION}-windows.msi /quiet'"
   elif doIsMacOs
   then
     doFail "Sorry, AWS CLI installation support is not yet implemented for your OS. Please install manually or help devonfw-ide to support it for your OS by contributing a pull-request."
   else
     doEcho "Installating AWS for Linux..."
     doRunCommand "sudo ${path_to_package}/install"
   fi
-  doRunCommand "rm -rf ${path_to_package}"
+  doRunCommand "rm -rf ${path_to_package}"
```
Comment (hohwille) [inline scripts/src/main/resources/scripts/command/aws:54]: accidental change to add trailing spaces?

File: scripts/src/main/resources/scripts/command/aws [hunk 4]
```diff
@@ -29,43 +43,46 @@ function doPackageInstall()
   then
     doEcho "Installing AWS for Windows..."
     windows_path_to_package=$(cygpath -w "${path_to_package}")
-    powershell.exe -Command "Start-Process msiexec.exe -verb runas -Wait -ArgumentList '/I ${windows_path_to_package}\aws-${current_version}-AWSCLIV2.msi /quiet'"
+    powershell.exe -Command "Start-Process msiexec.exe -verb runas -Wait -ArgumentList '/I ${windows_path_to_package}\aws-${AWS_VERSION}-windows.msi /quiet'"
```
Comment (hohwille) [inline scripts/src/main/resources/scripts/command/aws:46]: function is not really reusable as `${AWS_VERSION}` has to be initialized what happens inside `doSetup`.
Unfortunately bash has no concept of visibilities like `private` vs. `public`.
However, I understood that you extracted this function to increase readability (`doSetup` gets shorter and this "package installation" is grouped in a function with a self explanatory name. So all fine...

**Review by hohwille — CHANGES_REQUESTED** [review]
@CREITZ25 Thanks for testing, finding bugs and fixing. 👍 
Still I have some questions/doubts... Can you have a look. Lets also try to do a short call together...

File: scripts/src/main/resources/scripts/command/aws (hunk 3 above)
Comment (CREITZ25) [reply scripts/src/main/resources/scripts/command/aws:54]: I changed this line multiple times to see what happend by running the script.  I will delete the spaces again.

File: scripts/src/main/resources/scripts/command/aws (hunk 4 above)
Comment (CREITZ25) [reply scripts/src/main/resources/scripts/command/aws:46]: Now I add the version as the second parameter. The better way.

File: scripts/src/main/resources/scripts/command/aws (hunk 2 above)
Comment (CREITZ25) [reply scripts/src/main/resources/scripts/command/aws:82]: It misunderstood the word "silent". Now I corrected it

File: scripts/src/main/resources/scripts/command/aws (hunk 4 above)
Comment (hohwille) [reply scripts/src/main/resources/scripts/command/aws]: nice solution. Thanks.

**Review by hohwille — APPROVED** [review]
@CREITZ25 thanks for addressing all review comments perfectly. 👍 
Ready for merge.

**Pull request merged**